src/ngraph/runtime/cpu/cpu_external_function.cpp

```diff
@@ -1703,6 +1703,11 @@ shared_ptr<ngraph::runtime::cpu::CPU_CallFrame>
     runtime::cpu::CPU_ExternalFunction::make_call_frame(ngraph::pass::PassConfig& pass_config)
 {
 #if !defined(NGRAPH_DEX_ONLY)
+    // Override DEX if pass_config requests CODEGEN
+    if (pass_config.get_compilation_mode() == ngraph::pass::CompilationMode::CODEGEN)
+    {
+        m_direct_execution = false;
```

Comment:
To prevent unexpected behavior, shouldn't we better assert here? Like "CodeGen mode not supported"?

Reply:
In my mind, pass_config is requests from the user that we will do our best to adhere to. May be a warning instead of assert?

Reply:
Sure! It should be fine as long as the user can see "hey, you said CG but I'm running DEX instead".